pkg/cmd/server/api/validation/ldap.go

```diff
@@ -185,6 +191,13 @@ func ValidateLDAPQuery(query api.LDAPQuery, fldPath *field.Path) ValidationResul
 			"timeout must be equal to or greater than zero"))
 	}
 
+	if isDNOnly {
+		if len(query.Filter) != 0 {
+			validationResults.AddErrors(field.Invalid(fldPath.Child("filter"), query.Filter, `cannot specify a filter when using "dn" as the UID attribute`))
+		}
+		return validationResults
+	}
+
 	if _, err := ldap.CompileFilter(query.Filter); err != nil {
```

Comment:
This implicitly disallows not specifying a filter, so how is the validation not failing for those missing filters?